src/plan_expand_hypertable.c

```diff
@@ -0,0 +1,211 @@
+#include <optimizer/restrictinfo.h>
+#include <catalog/pg_inherits_fn.h>
+#include <nodes/plannodes.h>
+#include <optimizer/prep.h>
+
+#include "plan_expand_hypertable.h"
+#include "hypertable.h"
+#include "hypertable_restrict_info.h"
+#include "planner_import.h"
+
+
+/* Since baserestrictinfo is not yet set by the planner, we have to derive
+ * it ourselves. It's safe for us to miss some restrict info clauses (this
+ * will just results in more chunks being included) so this does not need
+ * to be as comprehensive as the PG native derivation.*/
+static List *
+getRestrictInfo(PlannerInfo *root, RelOptInfo *rel)
+{
+	FromExpr   *f = root->parse->jointree;
+	ListCell   *lc;
+	List	   *result = NIL;
+
+	if (f == NULL)
+		return NIL;
+
+	foreach(lc, (List *) f->quals)
+	{
+		Node	   *qual = (Node *) lfirst(lc);
+		RestrictInfo *restrictinfo;
+
+		Relids		relids = pull_varnos(qual);
+
+		if (bms_num_members(relids) != 1 || !bms_is_member(rel->relid, relids))
+			continue;
+
+		restrictinfo = make_restrictinfo((Expr *) qual,
+										 true,
+										 false,
+										 false,
+										 root->qual_security_level,
+										 relids,
+										 NULL,
+										 NULL);
+		result = lappend(result, restrictinfo);
+	}
+	return result;
+}
+
+static List *
+find_children_oids(HypertableRestrictInfo *hri, Hypertable *ht, LOCKMODE lockmode)
+{
+	/*
+	 * optimization: using the HRI only makes sense if we ar not using all the
+	 * chunks, otherwise using the cached inheritance hierarchy is faster.
+	 */
+	if (hypertable_restrict_info_has_restrictions(hri))
+	{
+		/* always include parent again, just as find_all_inheritors does */
+		List	   *result = list_make1_oid(ht->main_table_relid);
+
+		/* add chunks */
+		result = list_concat(result,
+							 hypertable_restrict_info_get_chunk_oids(hri,
+																	 ht,
+																	 lockmode));
+		return result;
+	}
+	else
+	{
+		return find_all_inheritors(ht->main_table_relid, lockmode, NULL);;
+	}
+}
+
+bool
+plan_expand_hypertable_valid_hypertable(Hypertable *ht, Query *parse, Index rti, RangeTblEntry *rte)
+{
+	if (ht == NULL ||
+	/* inheritance enabled */
+		rte->inh == false ||
+	/* row locks not necessary */
+		parse->rowMarks != NIL ||
+	/* not update and/or delete */
+		0 != parse->resultRelation)
+		return false;
+
+	return true;
+}
+
+/* Inspired by expand_inherited_rtentry but expands
+ * a hypertable chunks into an append rekationship */
+void
+plan_expand_hypertable_chunks(Hypertable *ht,
+							  PlannerInfo *root,
+							  Oid relationObjectId,
+							  bool inhparent,
+							  RelOptInfo *rel)
+{
+	RangeTblEntry *rte = rt_fetch(rel->relid, root->parse->rtable);
+	List	   *inhOIDs;
+	Oid			parentOID = relationObjectId;
+	ListCell   *l;
+	Relation	oldrelation = heap_open(parentOID, NoLock);
+	LOCKMODE	lockmode = AccessShareLock;
+	Query	   *parse = root->parse;
+	Index		rti = rel->relid;
+	List	   *appinfos = NIL;
+	HypertableRestrictInfo *hri;
+	PlanRowMark *oldrc;
```

Comment:
Unused variable (according to GCC on Ubuntu)

Reply:
actually nevermind, i see it getting used so I'm not sure why it complains.

Reply:
I guess because it's getting used in Assert and nowhere else it gets stripped out in Release builds. May need to use it in a no-op to get rid of the warning.